File: Python/agenda_queue.py
from collections import deque

def generate_queue(clients):
    queue = deque(sorted(clients, key=lambda c: c.appointment_date))
    return queue

def attend_queue(queue):
    print("\n--- Agenda Queue ---")
    while queue:
        client = queue.popleft()
        print("Attending:", client)

File: Java/AgendaQueue.java
import java.util.*;

public class AgendaQueue {
    private Queue<Client> queue = new LinkedList<>();

    public void generateQueue(List<Client> clients) {
        clients.stream()
                .sorted(Comparator.comparing(Client::getAppointmentDate))
                .forEach(queue::offer);
    }

    public void attendQueue() {
        System.out.println("\n--- Agenda Queue ---");
        while (!queue.isEmpty()) {
            System.out.println("Attending: " + queue.poll());
        }
    }
}
